src/Kernels/DeviceAux/hip/PlasticityAux.cpp

```diff
@@ -0,0 +1,256 @@
+#include <Kernels/DeviceAux/PlasticityAux.h>
+#include "hip/hip_runtime.h"
+#include <init.h>
+#include <cmath>
+
+#if REAL_SIZE == 8
+#define SQRT(X) sqrt(X)
+#define MAX(X,Y) fmax(X,Y)
+#elif REAL_SIZE == 4
+#define SQRT(X) sqrtf(X)
+#define MAX(X,Y) fmaxf(X,Y)
+#endif
```

Comment:
What do you think about this?
```
template<typename T>
__forceinline__ __device__ typename std::enable_if<std::is_floating_point<T>::value,T>::type
squareRoot(T x) {
  return std::is_same<T, double>::value ? sqrt(x) : sqrtf(x);
}

template<typename T>
__forceinline__ __device__ typename std::enable_if<std::is_floating_point<T>::value,T>::type
maxValue(T x, T y) {
  return std::is_same<T, double>::value ? fmax(x, y) : fmaxf(x, y);
}
```

Reply:
Why do we need to do this at all? Iirc std::sqrt has an overload for float/double anyway? Or are only the c-functions allowed?

Reply:
What I understood from the documentation of CUDA there are no overloads which is strange (https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html).

It is even more strange because cuda@11 has some concepts implemented from c++20